Sources/MapboxCoreNavigation/FixLocation.swift

```diff
@@ -2,13 +2,26 @@ import CoreLocation
 import Foundation
 import MapboxNavigationNative
 
-extension FixLocation {
 
+extension FixLocation {
     convenience init(_ location: CLLocation, isMock: Bool = false) {
         var bearingAccuracy: NSNumber? = nil
         if #available(iOS 13.4, *) {
             bearingAccuracy = location.courseAccuracy >= 0 ? location.courseAccuracy as NSNumber : nil
         }
+
+        var provider: String? = nil
+        #if compiler(>=5.5)
+        if #available(iOS 15.0, *) {
+            if let sourceInformation = location.sourceInformation {
+                // in some scenarios we store this information to history files, so to save space there, we use "short" names and 1/0 instead of true/false
+                let isSimulated = sourceInformation.isSimulatedBySoftware ? 1 : 0
+                let isProducedByAccessory = sourceInformation.isProducedByAccessory ? 1 : 0
+                
+                provider = "sim:\(isSimulated),acc:\(isProducedByAccessory)"
```

Comment:
Perhaps MapboxNavigationNative could be modified to accept something more structured like a dictionary instead of a freeform, user agent–like string, so that it can format and compress it as appropriate.

Reply:
I'm not sure we should add something iOS-specific - there is no analog of this structure on other platforms. Since this value will be used as kind of debug information I think it is okay to have it in this `provider` field in a free form. Btw initially I implemented it as a part of `FixLocation.extras` field, which allows to add any additional information to locations, but then decided that `provider` is exactly that field that was supposed to use for such use case(i.e. point to the source of location). WDYT?

Reply:
I meant a JSON-like dictionary of strings, similar to that `extras` field. But if you’re sure we won’t ever need additional structure beyond this format, then we can go with this more opaque string.

Reply:
I don't think we will need JSON here, I'd prefer to save bytes in history here rather than adding JSON overhead, because we likely need this data for debugging only.